lumen/base.py

```diff
@@ -39,13 +43,24 @@ def refs(self):
     def _get_type(cls, component_type):
         clsname = cls.__name__
         clslower = clsname.lower()
+        if component_type is None:
+            raise ValueError(f"'type' for '{clslower}' is not available.")
```

Comment:
Not sure this message is very clear:

```suggestion
            raise ValueError(f"No 'type' was provided during instantiation of '{clsname}' component.")
```

Reply:
Might use "materialization" over "instantiation" but the difference may be too technical. Materialization refers to the process of instantiating the object from a spec.